xontrib/brace_expansion.py

```diff
@@ -32,7 +36,11 @@ def expand_braces(cmd, **_):
                 prefix + option + postfix for option in options.split(",")
             ]
             expanded_cmds.extend(expanded_parts)
+            requires_expansion = True
         else:
             expanded_cmds.append(part)
 
-    return " ".join(expanded_cmds)
+    if requires_expansion:
+        return " ".join(expanded_cmds)
+
+    return cmd
```

Comment:
If we reach line 39, we always end up returning from line 44. There is no reachable code between these two points. Please remove `requires_expansion`.

Reply:
It would really be better to change one thing at a time to avoid having the core issue bogged down.